### DISCUSSION TITLE
in OsmAndMaps how does our audio note play Automatically

### DISCUSSION THREAD

**Discussion by AminSadighi** [osmandapp/OsmAnd#12773] [Maps Related]
Dear Sir
Hi,
How may we set osmandmaps, that it plays our recorded audio notes automatically, when we reach near favorites ( or reach near destination) in Android?

**Comment by vshcherb** [comment]
Unfortunately there is no functionality yet, it only plays sound that you are approaching favorites but doesn't play automatically them

**Comment by haeckerle** [comment]
I have made myself a simple demo of such a feature by adding an iteration through all audio recordings in SavingTrackHelper.updateLocation(). If Trip Recording plugin is enabled, it plays the recording based on proximity (i.e. if the distance is less than settings.SAVE_TRACK_PRECISION). 
![Screenshot_20221107-215835](https://user-images.githubusercontent.com/109791283/200416992-b3b4300c-ba6e-41ca-ac8f-3666968d61d3.png)
Works perfectly for me. Maybe it could be added as an option in the Audio Notes extension?

**Comment by AminSadighi** [comment]
Dear Sir,
Thank you very much
But I don't underestand how to set up it because the version of my osmand
is 4.2.7
If you creat a video clip for me or writing step by step in osmand menu, it
will be useful for me.
again thank you very much

RGDS
A.Sedighi

**Reply by haeckerle** [reply]
Dear Amin, 
the function is not available in the release yet. In order to test my demo, you would have to build OsmAnd with the additions proposed in [this commit](https://github.com/osmandapp/OsmAnd/commit/4077f0f3b3d3a11f08dec8e508268382a70f1ed2). Please refer to the [development setup](https://osmand.net/docs/technical/build-osmand/setup-the-dev-environment) and [build instructions for Android](https://osmand.net/docs/technical/build-osmand/how-to-compile-the-android-version).
